### PR TITLE
Add steps using  modern commanding

### PR DESCRIPTION
**Pull request by olegtk**
Update the walkthrough to also demonstrate how to handle commands using modern editor commanding added in 15.6.

### CODE REVIEW

File: docs/extensibility/walkthrough-using-a-shortcut-key-with-an-editor-extension.md [hunk 1]
```diff
@@ -41,8 +41,21 @@ You can respond to shortcut keys in your editor extension. The following walkthr
...
-## Defining the Command Filter  
+    ```csharp  
+    [Export(typeof(AdornmentLayerDefinition))]  
+    [Name("PurpleCornerBox")]  
+    [Order(After = PredefinedAdornmentLayers.Selection, Before = PredefinedAdornmentLayers.Text)]  
+    public AdornmentLayerDefinition editorAdornmentLayer;  
+    ```  
+
+## Handling TYPECHAR Command
+Prior to Visual Studio 2017 version 15.6 the only way to handle commands in an editor extention was implementing an <xref:Microsoft.VisualStudio.OLE.Interop.IOleCommandTarget> based command filter. Visual Studio 2017 version 15.6 introduced a modern simplified approach based on editor command handlers. The next two sections demonstarte how to handle a command using both legacy and modern approach.
```
Comment (gregvanl) [inline docs/extensibility/walkthrough-using-a-shortcut-key-with-an-editor-extension.md]: typos: extention -> extension, demonstarte -> demonstrate
"both the legacy and modern approach"

File: docs/extensibility/walkthrough-using-a-shortcut-key-with-an-editor-extension.md [hunk 2]
```diff
@@ -116,7 +129,7 @@ this.layer = view.GetAdornmentLayer("PurpleCornerBox");
   
     ```  
   
-## Adding the Command Filter  
+# Adding the Command Filter (prior to Visual Studio 2017 version 15.6)
```
Comment (gregvanl) [inline docs/extensibility/walkthrough-using-a-shortcut-key-with-an-editor-extension.md]: Needs to stay H2. Can't have two H1s.

File: docs/extensibility/walkthrough-using-a-shortcut-key-with-an-editor-extension.md [hunk 3]
```diff
@@ -183,7 +187,81 @@ this.layer = view.GetAdornmentLayer("PurpleCornerBox");
...
+    
+5.  Implement the `GetCommandState()` method as follows. Because this command handler handles core editor TYPECHAR command, it can delegate enabling the command to the core editor.
+  
+    ```csharp  
+    public CommandState GetCommandState(TypeCharCommandArgs args)
+    {
+        return CommandState.Unspecified;
+    } 
+    ```  
+
```
Comment (gregvanl) [inline docs/extensibility/walkthrough-using-a-shortcut-key-with-an-editor-extension.md]: jumped from step 5 to 7?

File: docs/extensibility/walkthrough-using-a-shortcut-key-with-an-editor-extension.md [hunk 4]
```diff
@@ -41,8 +41,21 @@ You can respond to shortcut keys in your editor extension. The following walkthr
...
+    [Export(typeof(AdornmentLayerDefinition))]  
+    [Name("PurpleCornerBox")]  
+    [Order(After = PredefinedAdornmentLayers.Selection, Before = PredefinedAdornmentLayers.Text)]  
+    public AdornmentLayerDefinition editorAdornmentLayer;  
+    ```  
+
+## Handling TYPECHAR Command
+Prior to Visual Studio 2017 version 15.6 the only way to handle commands in an editor extention was implementing an <xref:Microsoft.VisualStudio.OLE.Interop.IOleCommandTarget> based command filter. Visual Studio 2017 version 15.6 introduced a modern simplified approach based on editor command handlers. The next two sections demonstarte how to handle a command using both legacy and modern approach.
+
+# Defining the Command Filter (prior to Visual Studio 2017 version 15.6)
```
Comment (gregvanl) [inline docs/extensibility/walkthrough-using-a-shortcut-key-with-an-editor-extension.md]: Make H2, only 1 H1 per topic.

**Comment by PRMerger15** [comment]
@olegtk : Thanks for your contribution! The author, @gregvanl, has been notified to review your proposed change.

**Comment by gregvanl** [comment]
#sign-off

**Pull request merged**